packages/lodestar/src/network/util.ts

```diff
@@ -133,12 +133,14 @@ export async function dialProtocol(
   libp2p: LibP2p,
   peerId: PeerId,
   protocol: string,
-  timeout: number
+  timeout: number,
+  signal?: AbortSignal
 ): ReturnType<LibP2p["dialProtocol"]> {
   const abortController = new AbortController();
   const timer = setTimeout(() => {
     abortController.abort();
   }, timeout);
+  signal?.addEventListener("abort", () => clearTimeout(timer), {once: true});
```

Comment:
I don't think this is right.
If the reqResp controller aborts, we want to also abort any in-progress dials.
Currently, this would STOP the aborting of any in-progress dials in that case (by clearing the timeout that would otherwise abort them).

So this listener should be something like:
```suggestion
  signal?.addEventListener("abort", () => abortController.abort(), {once: true});
```